latch_cli/snakemake/workflow.py

```diff
@@ -0,0 +1,490 @@
+import importlib
+import typing
+from collections import OrderedDict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Type, TypeVar, Union
+
+import snakemake
+from flytekit.configuration import SerializationSettings
+from flytekit.core import constants as _common_constants
+from flytekit.core.class_based_resolver import ClassStorageTaskResolver
+from flytekit.core.docstring import Docstring
+from flytekit.core.interface import Interface, transform_interface_to_typed_interface
+from flytekit.core.node import Node
+from flytekit.core.promise import NodeOutput, Promise
+from flytekit.core.python_auto_container import (
+    DefaultTaskResolver,
+    PythonAutoContainerTask,
+)
+from flytekit.core.type_engine import TypeEngine
+from flytekit.core.workflow import (
+    WorkflowBase,
+    WorkflowFailurePolicy,
+    WorkflowMetadata,
+    WorkflowMetadataDefaults,
+)
+from flytekit.exceptions import scopes as exception_scopes
+from flytekit.models import interface as interface_models
+from flytekit.models import literals as literals_models
+from flytekit.models import types as type_models
+from snakemake.dag import DAG
+from snakemake.target_jobs import encode_target_jobs_cli_args
+
+from latch.types import LatchAuthor, LatchFile, LatchMetadata, LatchParameter
+
+SnakemakeInputVal = Union[snakemake.io._IOFile]
+
+
+def variable_name_for_file(file: snakemake.io.AnnotatedString):
```

Comment:
This can collapse two files into the same variable name.

Reply:
if by problematic characters, you meant the characters i am replacing with "_", the names I'm dealing with are file paths, so that wont work

will keep track of all file name -> variable name mappings wherever a collision could occur (for each job) and prefix collisions with a hash or something